src/Preconditions/ArrayPreconditions.php

```diff
@@ -0,0 +1,68 @@
+<?php namespace Guardsman\Preconditions;
+
+use Guardsman\Exceptions\KeyExists;
+use Guardsman\Exceptions\KeyNotFound;
+use Guardsman\Exceptions\ValueExists;
+use Guardsman\Exceptions\ValueNotFound;
+
+trait ArrayPreconditions
+{
+    /** @return mixed */
+    abstract public function getSubject();
+
+    /**
+     * @throws ValueNotFound if the subject is not a value within the array.
+     *
+     * @return $this
+     */
+    public function isValue(array $array)
```

Comment:
I can't help but wonder if this is the right way around. i.e `check($myArray)->isValue('foo')` What about checks such as `check($myArray)->isOnlyType('string')` or `check($myArray)->isOnlyClass(Foo::class)`.